### PR TITLE
fix: 128bit trace id propagation issue

### PR DESCRIPTION
**Pull request by codeboten**
When a service instrumented with ls-trace extracts a trace ID from a b3 header, if that ID is a 128 bit ID, the msgpack encoding fails. This change trims the ID if it is longer than 64 bits.

### CODE REVIEW

**Review by MitchellDumovic — APPROVED** [review]

File: ddtrace/span.py [hunk 1]
```diff
@@ -283,8 +283,12 @@ def get_metric(self, key):
         return self.metrics.get(key)
 
     def to_dict(self):
+        trace_id = self.trace_id
+        # ensure 128 bit trace IDs are always trimmed
+        if self.trace_id.bit_length() > 63:
```
Comment (ocelotl) [inline ddtrace/span.py:292]: Is this check really necessary? Can't we just do the operation below for any `trace_id`?

File: ddtrace/span.py [hunk 2]
```diff
@@ -283,8 +283,12 @@ def get_metric(self, key):
         return self.metrics.get(key)
 
     def to_dict(self):
+        trace_id = self.trace_id
+        # ensure 128 bit trace IDs are always trimmed
```
Comment (ocelotl) [inline ddtrace/span.py:291]: Is this comment accurate? It seems like any `trace_id` longer than 63 bits will be trimmed.

**Review by ocelotl — APPROVED** [review]
Minor comments :+1:

File: ddtrace/span.py (hunk 2 above)
Comment (codeboten) [reply ddtrace/span.py:291]: bit_length returns `63` for the max 64bit integer

File: ddtrace/span.py (hunk 1 above)
Comment (codeboten) [reply ddtrace/span.py:292]: we could, was trying to avoid unnecessary formatting

**Pull request merged**